### ISSUE TITLE
[Bug] SVTR以及SDMGR复现精度问题

### ISSUE THREAD

**Issue by GenerallyCovetous** [open-mmlab/mmocr#2058]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmocr/issues) and [Discussions](https://github.com/open-mmlab/mmocr/discussions) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (0.x)](https://github.com/open-mmlab/mmocr) or [latest version (1.x)](https://github.com/open-mmlab/mmocr/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

main branch https://github.com/open-mmlab/mmocr

### Environment

sys.platform: linux
Python: 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:53:27) [GCC 9.4.0]
CUDA available: True
MUSA available: False
numpy_random_seed: 2147483648
GCC: gcc (GCC) 7.3.0
PyTorch: 2.1.0
PyTorch compiling details: PyTorch built with:
  - GCC 10.2
  - C++ Version: 201703
  - Intel(R) MKL-DNN v3.1.1 (Git Hash 64f6bcbcbab628e96f33a62c3e975f8535a7bde4)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: NO AVX
  - Build settings: BLAS_INFO=open, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-10/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DLIBKINETO_NOROCTRACER -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-unused-private-field -Wno-aligned-allocation-unavailable -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=open, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.1.0, USE_CUDA=ON, USE_CUDNN=OFF, USE_EIGEN_FOR_BLAS=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=OFF, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.16.0
OpenCV: 4.8.0
MMEngine: 0.10.4
MMOCR: 1.0.1+

### Reproduces the problem - code sample

python tools/train.py configs/textrecog/master/master_resnet31_12e_st_mj_sa.py 

### Reproduces the problem - command or script

python tools/train.py configs/textrecog/master/master_resnet31_12e_st_mj_sa.py 

### Reproduces the problem - error message

[Attention] The same custom dataset val with master model seems normal, while svtr extremely low.
Epoch(train)  [1][14000/14111]  lr: 9.3503e-05  eta: 1 day, 8:06:25  time: 0.8898  data_time: 0.3493  loss: 0.2368  loss_ctc: 0.2368
2024/07/22 21:38:20 - mmengine - INFO - Exp name: svtr-tiny_20e_st_mj_20240722_180223
2024/07/22 21:38:20 - mmengine - INFO - Saving checkpoint at 1 epochs
2024/07/22 21:38:51 - mmengine - INFO - Epoch(val) [1][55/55]    CUSTOM/recog/word_acc: 0.0366  CUSTOM/recog/word_acc_ignore_case: 0.3671  CUSTOM/recog/word_acc_ignore_case_symbol: 0.3671  CUSTOM/recog/char_recall: 0.6272  CUSTOM/recog/char_precision: 0.7640  data_time: 0.1190  time: 0.5231
